### ISSUE TITLE
Modularisation of binDensity.py

### ISSUE THREAD

**Issue by eguil** [eguil/Density_bining#62]
- Organise `binDensity.py` with more external modules stored in libDensity.py
- move postpsro (mskAve...) into a new libDensityPospro.py
- modules to consider

1. maybe: interpolation proc (init and interp array)
2. switch model (e.g. masking of data, chunk calc,...): externalise ? 
3. l. 623 to 720 : binning kernel
4. interp l. 852-913
5. zonal mean ? l.931-962
6. persistence l. 966-1312
7. zona mean var init and write ? l. 1337-1388

**Comment by durack1** [comment]
@eguil I agree with this sentiment, if we can modularize things more so functions are reusable that would be great - it would also aid multiprocessing/threading these functions

**Comment by eguil** [comment]
libDensityPospro.py created

**Comment by eguil** [comment]
We need to decide how to distinguish global variables from local. We could use:
- `Capitalised` global vs.`lowercase` local
- `g_<var>` for global
- other suggestions, ideas, standards ?

**Comment by durack1** [comment]
@eguil how many regions do you want to define? I suggest we attempt to align as much with CF as we can http://cfconventions.org/Data/cf-standard-names/docs/standardized-region-names.html

**Comment by eguil** [comment]
no, not geographical regions ! -> names of python variables in the binDensity code

**Comment by durack1** [comment]
Ok got you, well what is the most important way to slice and dice variables in memory - having all the same quantities lining up together eg:
```
thetaoAtl
thetaoGlo
thetaoInd
thetaoPac
soAtl
...
```
Or the regional totals all lining up:
```
atlSo
atlThetao
gloSo
gloThetao
...
```

**Comment by eguil** [comment]
still not the issue ! Go and check the streamfunction branch created by Nicolas - you will see that he created variables in upper case (those that are coming from the libDensity.py). I find that upper case is really hard to read so we are looking for other ways of encoding this (first letter uppercase, g_<var>,...)

**Comment by durack1** [comment]
@eguil which file am I looking at https://github.com/eguil/Density_bining/blob/streamFunction/libDensity.py?

**Comment by eguil** [comment]
https://github.com/eguil/Density_bining/blob/streamFunction/binDensity.py

**Comment by durack1** [comment]
@eguil ok got you https://github.com/eguil/Density_bining/compare/master...streamFunction

I also agree that it would be nice to make the code more intuitively readable, some of the variable declarations don't make immediate sense to me. We really should attempt to leverage the [CMIP6 MIPvar names](http://clipc-services.ceda.ac.uk/dreq/index/CMORvar.html), but this will certainly come at a cost of length (e.g. thetao vs ptem)
